### PR TITLE
docs(rust): Add Rust example for tolerance join using join_asof in user guide

### PR DESCRIPTION
**Pull request by arnabanimesh**

### CODE REVIEW

File: docs/user-guide/transformations/joins.md [hunk 1]
```diff
@@ -218,11 +218,7 @@ If you want to make sure that only quotes within a certain time range are joined
 the `tolerance` argument. In this case we want to make sure that the last preceding quote is within 1 minute of the
 trade so we set `tolerance = "1m"`.
 
-=== ":fontawesome-brands-python: Python"
```
Comment (ritchie46) [inline docs/user-guide/transformations/joins.md:221]: I don't understand. Why is this removed?

Comment (arnabanimesh) [reply docs/user-guide/transformations/joins.md:221]: Python and Rust logos are already part of the auto-generated `code_block` template. No need for another Python logo.

Comment (ritchie46) [reply docs/user-guide/transformations/joins.md:221]: Alright, thanks!

**Review by ritchie46 — APPROVED** [review]

**Comment by arnabanimesh** [comment]
> Can you do a rebase? Then it's good to go.

Done

**Comment by arnabanimesh** [comment]
Fixed in PR #19127. Closing this PR.

**Comment by rodrigogiraoserrao** [comment]
Sorry for the overlapping work, @arnabanimesh, but I really needed to include the Rust example in the PR I made.
Thanks for your work 🚀

**Comment by arnabanimesh** [comment]
No problem @rodrigogiraoserrao. It happens. 😊
